src/physics/base/detail/PreStepActionImpl.hh

```diff
@@ -52,8 +52,7 @@ inline CELER_FUNCTION void pre_step_track(celeritas::CoreTrackView const& track)
 
     auto phys = track.make_physics_view();
     {
-        // Clear out energy deposition and secondary pointers even if the track
-        // is inactive
+        // Clear out energy deposition and secondary pointers
         phys.reset_energy_deposition();
         phys.secondaries({});
```

Comment:
OK, I thought I added code to EnergyDiagnostic to only apply to active tracks... but maybe I just had it check for nonzero energy deposition. I think we should move toward everything applying only to non-inactive tracks so we don't have to do the extra initialization of "unused" memory.

Related to that, one of the optimizations we'll probably want to do is to remap consecutive CUDA threads to disparate track slots (based on whether they're active/inactive, or what action ID is being launched) to reduce divergence and the number of blocks that have to be transported. At this point, most of our kernels will have different "threads" and "track thread IDs"—so perhaps we want to introduce a separate enum `SlotId` or `StateId` to indicate "index into the state array"  rather than "index of the CUDA array". That'll be a pretty large (mostly find/replace for now) change so definitely another pull request.

Reply:
Now that we're only resetting the energy deposition for active tracks we'll need to initialize it to zero at the start of the simulation, right? Also, since we're not clearing secondaries for inactive tracks anymore, won't we still be "processing" them in `ProcessSecondariesLauncher`?

Reply:
Or actually maybe we don't need to initialize the energy deposition. Just make sure we're only processing secondaries for active tracks?

Reply:
The energy deposition gets initialized to zero as part of the physics state, and we shouldn't be checking it if the track is inactive. And yeah good catch, `ProcessSecondariesLauncher` should only apply to inactive tracks :) so I will fix that.

Reply:
Also this makes me realize I need to mark a track as "alive" again if was re-initialized with a secondary after its death.

Reply:
OK fixed!